firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -271,6 +278,21 @@ String getName() {
     return Tasks.call(backgroundExecutor, this::deleteFirebaseInstallationId);
   }
 
+  /** Register a callback {@link FidListener} to receive fid changes. */
+  @NonNull
+  @Override
+  public synchronized HandleFidListener registerFidListener(@NonNull FidListener listener) {
+    this.fidListeners.add(listener);
+    return new HandleFidListener() {
+      @Override
+      public void unregister(FidListener fidListener) {
+        if (fidListeners.contains(fidListener)) {
+          fidListeners.remove(fidListener);
+        }
+      }
```

Comment:
While the registration is synchronized, the unregister is not, please consider something like
```suggestion
      public synchronized void unregister() {
        fidListeners.remove(listener);
      }
```